extensions/amp-video/0.1/video-cache.js

```diff
@@ -95,3 +104,34 @@ function applySourcesToVideo(videoEl, sources) {
       videoEl.insertBefore(sourceEl, videoEl.firstChild);
     });
 }
+
+/**
+ * Moves the src attribute to a source element to enable playing from multiple
+ * sources: the cached ones and the fallback initial src.
+ * @param {!Element} videoEl
+ * @param {!Window} win
+ * @return {!Promise}
+ */
+function replaceSrcWithSourceElement(videoEl, win) {
+  if (!videoEl.hasAttribute('src')) {
+    return;
+  }
+  const sourceEl = win.document.createElement('source');
```

Comment:
Optional: We have the `createElementWithAttributes` if you want to use it

Reply:
It makes it a bit harder to not set a `type` (vs setting it to `null` or `""`) if it was not on the `amp-video` in the first place, I'll keep the current implementation